### PR TITLE
[ONNX] enable several script unit tests using new jit passes

### PR DESCRIPTION
**Pull request by hwangdeyu**
Tests that have been enabled include:
### Passed
- test_interpolate_upsample
- test_interpolate_function_substitution
- test_interpolate_downsample
- test_interpolate_no_shape
- test_lstm
- test_lstm_default_init_state
- test_lstm_fixed_batch_size
- test_lstm_constant_folding
- test_lstm_no_bias
- test_einsum
- test_lstm_post_fix_init_state
- test_binary_cross_entropy_with_logits

### Fixed annotation type, but still failed.
- test_rnn_no_bias 
- test_pad_types
- test_roi_heads
- test_multi_scale_roi_align
- test_rpn
- test_transform_images
- test_clip_boxes_to_image

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -1964,35 +1964,6 @@ def forward(self, x):
...
-        class MyModel(torch.nn.Module):
-            def forward(self, x):
-                scale = 2.3 if is_upsample else 0.5
-                if len(x.size()) == 3:
-                    scale_array = 2.3
-                if len(x.size()) == 4:
-                    scale_array = [2.3, 5.1]
-                if len(x.size()) == 5:
-                    scale_array = [3.3, 2.3, 5.1]
-                if use_size:
```
Comment (jiafatom) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:1977]: As you delete this model, the question is: Is this `use_size=True` path covered in the UT?

Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:1977]: Yes, this part is a old usage which is used for exporting tracing test. Now the _interpolate_script default covered both scripting and tracing test, it's also covered use_size=True.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 2]
```diff
@@ -5507,30 +5475,38 @@ def forward(self, input, target):
...
         self.run_test(NLLModel(), (input, target))
 
-    @disableScriptTest()
     @skipIfUnsupportedMinOpsetVersion(12)
     def test_binary_cross_entropy_with_logits(self):
         x = torch.randn(5)
         y = torch.empty(5).random_(2)
         self._bce_logits_loss(x, y)
 
-        x = torch.randn(2, 3, 5, 7)
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:5517]: We need 4 variants of this test:
bce_logits
bce_logits_wegiht
bce_logits_posweight
bce_logits_weight_posweight

Please do not delete any tests. This is critical.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 3]
```diff
@@ -1964,35 +1964,6 @@ def forward(self, x):
...
-                        torch.nn.functional.interpolate(x, mode=mode, scale_factor=scale_array,
-                                                        align_corners=True, recompute_scale_factor=False)
-                return torch.nn.functional.interpolate(x, mode=mode,
-                                                       scale_factor=scale, recompute_scale_factor=False), \
-                    torch.nn.functional.interpolate(x, mode=mode,
-                                                    scale_factor=scale_array, recompute_scale_factor=False)
-
-        self.run_test(MyModel(), x)
-
     def _interpolate_script(self, x, mode, use_size, is_upsample, align_corners=False):
```
[the remark this answers is not in the document]
Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py]: Thanks, these two test do the same thing, fixed it.

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 4]
```diff
@@ -2110,23 +2080,19 @@ def _interpolate_tests(self, is_upsample):
...
+                    self._interpolate(xi, mode_i, True, is_upsample)
                     # test with align_corners if supported
                     if mode != 'nearest':
                         self._interpolate(xi, mode_i, False, is_upsample, True)
-                        self._interpolate_script(xi, mode_i, False, is_upsample, True)
-                    self._interpolate_script(xi, mode_i, False, is_upsample)
+                    self._interpolate(xi, mode_i, False, is_upsample)
 
-    @disableScriptTest()
+    @skipIfUnsupportedMinOpsetVersion(9)
```
Comment (neginraoof) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:2092]: Why is this failing for opsets < 9?

Comment (hwangdeyu) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:2093]: Thank you, have added the comments.

**Review by neginraoof — APPROVED** [review]
Thanks a lot, looks good to me!

**Pull request merged**